packages/wallet/src/redux/protocols/existing-channel-funding/reducer.ts

```diff
@@ -271,46 +267,36 @@ const waitForLedgerUpdateReducer = (
   return { protocolState: states.waitForPostFundSetup(protocolState), sharedData: newSharedData };
 };
 
-function craftNewAllocationAndDestination(
+function ledgerChannelNeedsTopUp(
   latestCommitment: Commitment,
-  proposedAmount: string,
-  channelId: string,
-): { proposedAllocation: string[]; proposedDestination: string[] } {
-  const numParticipants = helpers.getNumberOfParticipants(latestCommitment);
-  const amountRequiredFromEachParticipant = bigNumberify(proposedAmount).div(numParticipants);
-
-  const proposedAllocation: string[] = [];
-  const proposedDestination: string[] = [];
-
-  for (let i = 0; i < latestCommitment.allocation.length; i++) {
-    const allocation = latestCommitment.allocation[i];
-
-    const newAmount = bigNumberify(allocation).sub(amountRequiredFromEachParticipant);
-
-    if (newAmount.gt('0x0')) {
-      proposedAllocation.push(newAmount.toHexString());
-      proposedDestination.push(latestCommitment.destination[i]);
-    }
-  }
-
-  proposedAllocation.push(proposedAmount);
-  proposedDestination.push(channelId);
-
-  return { proposedAllocation, proposedDestination };
-}
-
-function ledgerChannelNeedsTopUp(latestCommitment: Commitment, proposedAmount: string) {
+  proposedAllocation: string[],
+  proposedDestination: string[],
+) {
   if (latestCommitment.commitmentType !== CommitmentType.App) {
     throw new Error('Ledger channel is already closed.');
   }
-  const numParticipants = helpers.getNumberOfParticipants(latestCommitment);
-  const amountRequiredFromEachParticipant = bigNumberify(proposedAmount).div(numParticipants);
-
-  return !latestCommitment.allocation.every(a =>
-    bigNumberify(a).gte(amountRequiredFromEachParticipant),
-  );
+  // We assume that destination/allocation are the same length and destination contains the same addresses
+  // Otherwise we shouldn't be in this protocol at all
+  for (let i = 0; i < proposedDestination.length; i++) {
```

Comment:
This could be something like
```
return proposedDestination.some(address => {
  const existingIndex = latestCommitment.destination.indexOf(address);
  return bigNumberify(latestCommitment.allocation[existingIndex]).lt(proposedAllocation[i]);
}
``